inflation/lp/InflationLP.py

```diff
@@ -0,0 +1,1862 @@
+            else:
+                CG_nonadjusting_ortho_groups_as_boolarrays.append(
+                    ortho_groups_as_boolarrays)
+
+        self.CG_adjusting_ortho_groups_as_boolarrays = list(chain.from_iterable(CG_adjusting_ortho_groups_as_boolarrays))
+        self.CG_nonadjusting_ortho_groups_as_boolarrays = list(chain.from_iterable(CG_nonadjusting_ortho_groups_as_boolarrays))
+        self.all_ortho_groups_as_boolarrays = list(chain.from_iterable(all_ortho_groups_as_boolarrays))
+
+        # We want to consider ALL outcomes for variables which have children, 
+        # but not the last outcome for childless variables.
+        # In other words, if we take all the CG equalities, we want to convert 
+        # them to inequalities when the LHS is non_CG but childless-only.
+        if np.any(self.does_not_have_children):
+            bad_boolvecs_for_ineqs = [bool_array[-1] for bool_array in self.CG_adjusting_ortho_groups_as_boolarrays]
+            self.boolvec_for_CG_ineqs = np.bitwise_or.reduce(bad_boolvecs_for_ineqs, axis=0)
+        else:
+            self.boolvec_for_CG_ineqs = self.blank_bool_vec
+        if np.any(self.has_children):
+            bad_boolvecs_for_eqs = [bool_array[-1] for bool_array in self.CG_nonadjusting_ortho_groups_as_boolarrays]
+            self.boolvec_for_FR_eqs = np.bitwise_or.reduce(bad_boolvecs_for_eqs, axis=0)
+        else:
+            self.boolvec_for_FR_eqs = self.blank_bool_vec
+
+        if self.verbose > 1:
+            print("Number of single operator measurements per party:", end="")
+            prefix = " "
+            for i, measures in enumerate(inflationproblem.measurements):
+                op_count = np.prod(measures.shape[:2])
+                print(prefix + f"{self.names[i]}={op_count}", end="")
+                prefix = ", "
+            print()
+        self.use_lpi_constraints = False
+
+        self.identity_operator = np.empty((0, self._nr_properties),
+                                          dtype=self.np_dtype)
+        self.zero_operator = np.zeros((1, self._nr_properties),
+                                      dtype=self.np_dtype)
+
+        # These next properties are reset during generate_lp, but are needed in
+        # init so as to be able to test the Monomial constructor function
+        # without generate_lp.
+        #TODO: Enable test of Monomial constructor without generate_lp!
+        self.atomic_monomial_from_hash  = dict()
+        self.monomial_from_atoms        = dict()
+        self.monomial_from_name         = dict()
+        self.One  = self.Monomial(self.identity_operator, idx=1)
+        self._generate_lp()
+
+    ###########################################################################
+    # MAIN ROUTINES EXPOSED TO THE USER                                       #
+    ###########################################################################
+    def _generate_lp(self) -> None:
```

Comment:
@Elie Wolfe rename to generate_relaxation() ?
```suggestion
    def _generate_lp(self) -> None:
```